From 4bf7b0d2be1f5e0f17338db3b1a522b53bf30571 Mon Sep 17 00:00:00 2001
From: Florin Sarbu <florin@resin.io>
Date: Fri, 20 Apr 2018 07:28:45 +0200
Subject: [PATCH] overlayfs: Fix calls to vfs_rename() vfs_unlink()

We backported some patches that changed the prototypes of vfs_rename() and
vfs_unlink() so we need to align here with those changes.

Upstream-Status: Backport
Signed-off-by: Florin Sarbu <florin@resin.io>
---
 fs/overlayfs/copy_up.c | 2 +-
 fs/overlayfs/dir.c     | 8 ++++----
 fs/overlayfs/readdir.c | 2 +-
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/fs/overlayfs/copy_up.c b/fs/overlayfs/copy_up.c
index 7b9fb6d..2c26232 100644
--- a/fs/overlayfs/copy_up.c
+++ b/fs/overlayfs/copy_up.c
@@ -228,7 +228,7 @@ err_remove:
 	if (S_ISDIR(stat->mode))
 		vfs_rmdir(upperdir->d_inode, newpath.dentry);
 	else
-		vfs_unlink(upperdir->d_inode, newpath.dentry);
+		vfs_unlink(upperdir->d_inode, newpath.dentry, NULL);
 
 	dput(newpath.dentry);
 
diff --git a/fs/overlayfs/dir.c b/fs/overlayfs/dir.c
index f4969c1..ede0889 100644
--- a/fs/overlayfs/dir.c
+++ b/fs/overlayfs/dir.c
@@ -59,7 +59,7 @@ static int ovl_whiteout(struct dentry *upperdir, struct dentry *dentry)
 
 	err = vfs_setxattr(newdentry, ovl_whiteout_xattr, "y", 1, 0);
 	if (err)
-		vfs_unlink(upperdir->d_inode, newdentry);
+		vfs_unlink(upperdir->d_inode, newdentry, NULL);
 
 out_dput:
 	dput(newdentry);
@@ -117,7 +117,7 @@ static struct dentry *ovl_lookup_create(struct dentry *upperdir,
 
 		err = -EEXIST;
 		if (ovl_is_whiteout(newdentry))
-			err = vfs_unlink(upperdir->d_inode, newdentry);
+			err = vfs_unlink(upperdir->d_inode, newdentry, NULL);
 
 		revert_creds(old_cred);
 		put_cred(override_cred);
@@ -367,7 +367,7 @@ static int ovl_do_remove(struct dentry *dentry, bool is_dir)
 		if (is_dir)
 			err = vfs_rmdir(upperdir->d_inode, realpath.dentry);
 		else
-			err = vfs_unlink(upperdir->d_inode, realpath.dentry);
+			err = vfs_unlink(upperdir->d_inode, realpath.dentry, NULL);
 		if (err)
 			goto out_d_drop;
 
@@ -557,7 +557,7 @@ static int ovl_rename(struct inode *olddir, struct dentry *old,
 	}
 
 	err = vfs_rename(old_upperdir->d_inode, olddentry,
-			 new_upperdir->d_inode, newdentry);
+			 new_upperdir->d_inode, newdentry, NULL);
 
 	if (err) {
 		if (new_create && ovl_dentry_is_opaque(new))
diff --git a/fs/overlayfs/readdir.c b/fs/overlayfs/readdir.c
index 4c18abf..bae0baa 100644
--- a/fs/overlayfs/readdir.c
+++ b/fs/overlayfs/readdir.c
@@ -535,7 +535,7 @@ static int ovl_remove_whiteouts(struct dentry *dir, struct list_head *list)
 			    p->len, p->name, PTR_ERR(dentry));
 			continue;
 		}
-		ret = vfs_unlink(upperdir->d_inode, dentry);
+		ret = vfs_unlink(upperdir->d_inode, dentry, NULL);
 		dput(dentry);
 		if (ret)
 			pr_warn(
-- 
2.7.4

